wgpu-hal/src/dx12/command.rs

```diff
@@ -30,10 +30,7 @@ impl crate::BufferTextureCopy {
                 )
                 .unwrap(),
                 Width: self.size.width,
-                Height: self
-                    .buffer_layout
-                    .rows_per_image
-                    .map_or(self.size.height, |count| count * block_height),
+                Height: self.size.height,
```

Comment:
This looks correct but I'm not sure what happens if you copy a 3D texture with a non-default `rows_per_image`.
We have `RowPitch` but no `ColPitch`.
Vulkan has [`bufferImageHeight`](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkBufferImageCopy.html) and Metal has [`sourceBytesPerImage`](https://developer.apple.com/documentation/metal/mtlblitcommandencoder/1400752-copyfrombuffer).

@cwfitzgerald any ideas?

Reply:
Dawn appears to issue one copy per layer, though I'm not 100% sure on this, as dawn has a lot of layers to their d3d12 copy code.

It seems like we should issue one copy if it's dense, otherwise one copy per subresource if it's not

Reply:
That's my conclusion as well, Vulkan and Metal support specifying `rows_per_image` for 3D textures but D3D12 doesn't.
So, for a non-default `rows_per_image` we should set `depth: 1` and issue multiple copies by advancing `D3D12_PLACED_SUBRESOURCE_FOOTPRINT.Offset` by `rows_per_image * depth_index`.

This is a pre-existing issue, will open an issue to track it.